sources/cfsignal/src/cloudformation.rs

```diff
@@ -0,0 +1,65 @@
+use crate::error::{self, Result};
+use imdsclient::ImdsClient;
+use log::info;
+use rusoto_cloudformation::{
+    CloudFormation, CloudFormationClient, SignalResourceInput,
+};
+use rusoto_core::Region;
+use snafu::OptionExt;
+use snafu::ResultExt;
+use std::str::FromStr;
+
+pub(crate) struct SignalResource {}
+
+impl SignalResource {
+    #[tokio::main]
+    pub(crate) async fn signal_resource(
```

Comment:
Since we don't do anything with the `SignalResource` struct, it'd be better to just use a function - then we can call `signal_resource(...)` instead of `SignalResource::signal_resource(...)`.

It's also a little odd to bury `#[tokio::main]` in a module like this - ideally it'd be on the `main` function.